### PR TITLE
Parsing anchors & aliases

### PR DESCRIPTION
**Pull request by kpodsiad**
We still cannot parse anchor when a node is mapping key at the same time (require tokenizer changes from #87). Otherwise, it should just work. 

Follow-up: Handle anchors & aliases when composing nodes https://github.com/VirtusLab/scala-yaml/issues/88

### CODE REVIEW

File: yaml/shared/src/main/scala/org/virtuslab/yaml/internal/load/compose/Composer.scala [hunk 1]
```diff
@@ -35,7 +35,9 @@ object ComposerImpl extends Composer:
         case _: Event.SequenceStart                            => composeSequenceNode(tail)
         case _: Event.MappingStart | _: Event.FlowMappingStart => composeMappingNode(tail)
         case s: Event.Scalar                                   => composeScalarNode(s, tail)
-        case event => Left(ComposerError(s"Unexpected event $event"))
+        // todo #88
```
Comment (lwronski) [inline yaml/shared/src/main/scala/org/virtuslab/yaml/internal/load/compose/Composer.scala:38]: Add more descriptive description

File: yaml/shared/src/main/scala/org/virtuslab/yaml/internal/load/parse/ParserImpl.scala [hunk 2]
```diff
@@ -109,6 +105,7 @@ final class ParserImpl private (in: Tokenizer) extends Parser:
     def loop(events: mutable.ArrayDeque[Event]): Either[YamlError, List[Event]] = {
       getNextEvent() match
         case Right(event) =>
+          println(event)
```
Comment (lwronski) [inline yaml/shared/src/main/scala/org/virtuslab/yaml/internal/load/parse/ParserImpl.scala:108]: To remove

File: yaml/shared/src/main/scala/org/virtuslab/yaml/internal/load/parse/Event.scala [hunk 3]
```diff
@@ -29,18 +30,25 @@ object Event:
...
+      extends Sequence
+  case class SequenceEnd(pos: Option[Position] = None) extends Sequence
 
-  sealed trait Mapping                                      extends Node
-  case class MappingStart(pos: Option[Position] = None)     extends Mapping
-  case class MappingEnd(pos: Option[Position] = None)       extends Mapping
-  case class FlowMappingStart(pos: Option[Position] = None) extends Mapping
-  case class FlowMappingEnd(pos: Option[Position] = None)   extends Mapping
+  sealed trait Mapping extends Node
+  case class MappingStart(pos: Option[Position] = None, anchor: Option[String] = None)
```
Comment (lwronski) [inline yaml/shared/src/main/scala/org/virtuslab/yaml/internal/load/parse/Event.scala:49]: I think, we should add Context field, which contains for example `pos` or `anchor`

File: yaml/shared/src/main/scala/org/virtuslab/yaml/internal/load/parse/ParserImpl.scala [hunk 4]
```diff
@@ -362,21 +327,51 @@ final class ParserImpl private (in: Tokenizer) extends Parser:
...
+      nextToken.kind match
+        case TokenKind.Scalar(value, style) =>
+          in.popToken()
+          Right(Event.Scalar(value, style, Some(pos), anchor))
+        case _ =>
+          Left(ParseError.from(TokenKind.Scalar.toString, token))
+
+    inline def parseFlowNode() = parseNode(couldParseBlockColl = false)
+
+    def parseNode(couldParseBlockColl: Boolean = true): Either[YamlError, Event] =
```
Comment (lwronski) [inline yaml/shared/src/main/scala/org/virtuslab/yaml/internal/load/parse/ParserImpl.scala:341]:
```suggestion
    def parseNode(couldParseBlockCollection: Boolean = true): Either[YamlError, Event] =
```

**Review by lwronski — APPROVED** [review]
LGTM

File: yaml/shared/src/main/scala/org/virtuslab/yaml/internal/load/parse/Event.scala (hunk 3 above)
Comment (kpodsiad) [reply yaml/shared/src/main/scala/org/virtuslab/yaml/internal/load/parse/Event.scala:49]: I'll do this later, in another PR.

**Pull request merged**